normandy/recipes/filters.py

```diff
@@ -224,6 +224,98 @@ def capabilities(self):
         return set()
 
 
+class AddonActiveFilter(BaseFilter):
+    """Match a user based on if a particular addon is active.
+
+    .. attribute:: type
+
+        ``addon_active``
+
+    .. attribute:: addons
+        List of addon ids to filter against.
+
+        :example: ``["uBlock0@raymondhill.net", "pioneer-opt-in@mozilla.org"]``
+
+    .. attribute:: any_or_all
+        This will determine whether the addons are connected with an "&&" operator,
+        meaning all the addons must be active for the filter to evaluate to true,
+        or an "||" operator, meaning any of the addons can be active to evaluate to
+        true.
+
+        :example: ``any`` or ``all``
+    """
+
+    type = "addon_active"
+    addons = serializers.ListField(child=serializers.CharField(), min_length=1)
+    any_or_all = serializers.CharField()
+
+    def to_jexl(self):
+        any_or_all = self.initial_data["any_or_all"]
+
+        if any_or_all == "all":
+            return "&&".join(
+                (f'normandy.addons["{addon}"].isActive' for addon in self.initial_data["addons"])
+            )
+        elif any_or_all == "any":
+            return "||".join(
+                (f'normandy.addons["{addon}"].isActive' for addon in self.initial_data["addons"])
+            )
+        else:
+            raise serializers.ValidationError(
+                f"Unrecognized string for any_or_all: {any_or_all!r}"
+            )
```

Comment:
probably a cleaner way to do this is:
```py
if any_or_all == "all":
    glue = "&&"
elif any_or_all == "any":
    glue = "||"
else:
    raise # Exception here

return glue.join(f'...')
```

Reply:
yes you're right

Reply:
Not to butt in but generally for this kind of mapping pattern you can use a dictionary